### ISSUE TITLE
Allow commenting on collapsed lines in PR

### ISSUE THREAD

**Issue by kanelbulle** [isaacs/github#1655]
In a PR it seems you can only comment on the lines that are "close" to the changed lines. This is not always sufficient, since comments can be used to point out lines that will have to be changed in order to correct an issue.

For example, if in a PR you open a collapsed section of code, it seems impossible to leave a comment on any of the collapsed lines (happy to be proven wrong on this).

**Comment by frank-weindel** [comment]
This is a duplicate of #284

**Comment by kanelbulle** [comment]
Indeed, thanks!

**Comment by alkhimey** [comment]
Any chance this issue will be addressed? I find myself commenting on arbitrary lines with notes like  `this comment actually refers to line #1337:`.

**Comment by vimalk78** [comment]
same here. allow comments on lines far from changed lines

**Comment by gurayAlsac** [comment]
+1 Sometimes a reviewer needs to point out a section that needs changing but hasn't been addressed in the PR.

**Comment by grabear** [comment]
+1

**Comment by agolynski** [comment]
+1

**Comment by brendadeely** [comment]
Agreed with all of the above. Plus, if a commit isn't quite right, a commenter may want to suggest changes in a comment to lines that fall outside of the original commit. I personally have run into this a good dozen times at least in the last month or so.

**Comment by ledmonster** [comment]
I really want this feature. I can't understand why GitHub leaves this issue as it is. This is a fundamental feature for developers.

**Comment by irwineffect** [comment]
I suspect the current behavior was designed to help limit the scope of changes requested during a PR, but as others have mentioned, sometimes we really need to point out and add comments to other areas of the code that are directly related to the changes but aren't close to the changed lines.

**Comment by molshansky-zipwhip** [comment]
If we add the functionality, there's no reason it can't be configurable so folks that want/need it can use it and others that don't like the feature can turn it off (or perhaps it's off by default).

**Comment by felixpelletier** [comment]
My team does that extensively with another PR system. It's really a must have!

**Comment by FrankPortman** [comment]
I just used Github PR reviews for the first time (usually use Phabricator) and my mind was blown that this isn't possible. Would be a huge value-add imo!

**Comment by peterburnash** [comment]
A badly needed feature.

**Comment by preveenraj** [comment]
Any update on this issue? really need this!

**Comment by Arcanorum** [comment]
So annoying that this isn't possible.

**Comment by pb-julian** [comment]
Please add this feature to GitHub

**Comment by marxlow** [comment]
Bump. This is really important. Anything we can do to help speed this up?

**Comment by kosyakov-t** [comment]
How do I vote this up?  
I also need an ability to comment on the lines that are not within the "diff area" — to point to missed changes (where there should have been a change, but it was missed by the PR author).

**Comment by airlanggagunawan** [comment]
Yeah, came across this problem earlier today during a review. This feature would really help make things clearer.

**Comment by Sackbuoy** [comment]
Agreed. It would be helpful to be able to comment on any code in the file, not just what was add/changed

**Comment by DIRECTcut** [comment]
Was surprised that this is not possible. We need this feature

**Comment by nickrvieira** [comment]
Missing this feature as well.

**Comment by YuriiKovalenko-codeit** [comment]
+1

**Comment by sorob-raissi** [comment]
+5 Yup

**Comment by jesslacy** [comment]
+1

**Comment by kennethcyh** [comment]
+1

**Comment by RiverPlace** [comment]
+1

**Comment by Cuahchic** [comment]
+1

**Comment by renaudguerin** [comment]
Another +1

**Comment by Lafaa** [comment]
+1

**Comment by Hasenherz** [comment]
+1

**Comment by yelena-premise** [comment]
+1

**Comment by jtramm** [comment]
I agree -- this is a badly needed feature. Critical to have this ability when reviewing refactored code and you need to point out lines that were missed.

**Comment by ryomtoob** [comment]
+1

**Comment by NathanFoley** [comment]
+1

**Comment by nilesta** [comment]
+1

**Comment by nlakritz** [comment]
+1

**Comment by Levi-Lesches** [comment]
Can people stop leaving "+1" or "Me too" comments? Leave a 👍 on the issue and press "Subscribe" if you want notifications; commenting only notifies everyone watching this issue/repository. Also, GitHub does not monitor this repo so triggering notifications won't get their attention.

**Comment by ddellsperger** [comment]
It looks like @dotproto has suggested this on the new GitHub discussions [here](https://github.com/github/feedback/discussions/4452).  It would probably be worth subscribing there as well.

**Comment by mgf-bmanza** [comment]
+1

**Comment by davidvincze** [comment]
+1

**Comment by jc-hdez** [comment]
+1

**Comment by dotproto** [comment]
To support this idea, please visit upvote (not just +1) [this feature request](https://github.com/github/feedback/discussions/4452) on GitHub's feedback discussions page. A lot of people have used emoji reactions, but discussions use a separate vote system to rank requests. 

_Upvote button on the discussion detail view_
![Screen Shot 2021-07-15 at 5 43 05 PM](https://user-images.githubusercontent.com/1047763/125875227-5e01b860-11a3-403c-b405-201b4ce0b9f8.png)

_Upvote button on the the index page_
![Screen Shot 2021-07-15 at 5 43 50 PM](https://user-images.githubusercontent.com/1047763/125875114-b83ae694-9652-40a1-b3f0-508d76f81249.png)

Continuing to +1 here will be much less effective than engaging directly with the GitHub team.
